app/code/Magento/Cms/Model/Page/DataProvider.php

```diff
@@ -123,40 +107,71 @@ public function prepareMeta(array $meta)
      *
      * @return array
      */
-    public function getData()
+    public function getData(): array
     {
         if (isset($this->loadedData)) {
             return $this->loadedData;
         }
-        $this->collection = $this->collectionFactory->create();
-        $items = $this->collection->getItems();
-        /** @var $page \Magento\Cms\Model\Page */
-        foreach ($items as $page) {
-            $this->loadedData[$page->getId()] = $page->getData();
-            if ($page->getCustomLayoutUpdateXml() || $page->getLayoutUpdateXml()) {
-                //Deprecated layout update exists.
-                $this->loadedData[$page->getId()]['layout_update_selected'] = '_existing_';
-            }
+
+        $page = $this->getCurrentPage();
+        if ($page === null) {
+            return [];
+        }
```

Comment:
By adding exceptions to `getCurrentPage` we can wrap this on try-catch and handle the errors appropriately, what do you think?